### PR TITLE
Add user account details API

### PR DESCRIPTION
**Pull request by runlevel5**
This endpoint provides all details about a user given the username.

cc @fraserxu

### CODE REVIEW

File: lib/user.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+User.prototype = {
+  /**
+   * get user account details by username
+   * @param  {String} username
+   * @return {Promise}
+   */
+  getAccountDetails: function (username) {
+    if (!username) throw new Error('Username is required.')
+
+    return this._request.get('/v1/market/username:' + username + '.json')
```
Comment (fraserxu) [inline lib/user.js:18]: Is this the equivalent to the one we are using to talk to marketplace after sso login success?

**Comment by runlevel5** [comment]
this is a bit different. the response only contain non-sensitive details of the user.

**Comment by fraserxu** [comment]
Sorry I don't have enough context why we are making this api? Is this for the author suggestion?

**Comment by runlevel5** [comment]
@fraserxu yes

**Comment by fraserxu** [comment]
@joneslee85 Is the work being done on api-gateway side? I haven't seen a pr for that yet.

**Comment by runlevel5** [comment]
@fraserxu the delegation work is on the api-gateway side. There is no need PR on api-gateway

**Comment by fraserxu** [comment]
@joneslee85 Ah, ok, so this means api-gateway already has the `username` api so we don't need to do it right?

How does this related to author suggestion?

**Comment by runlevel5** [comment]
@fraserxu in order to retrieve the author profile image url, we need to call to this endpoint

**Review by fraserxu — APPROVED** [review]

**Pull request merged**